### ISSUE TITLE
Mypy cannot find torch.optim.AdamW

### ISSUE THREAD

**Issue by jeongukjae** [pytorch/pytorch#33757]
## 🐛 Bug

```python
from torch.optim import AdamW
```

Above code raises type error `test.py:1: error: Module 'torch.optim' has no attribute 'AdamW'; maybe "Adam"?`

## To Reproduce

1. Create new file and write this line `from torch.optim import AdamW`
1. Check with mypy

## Expected behavior

This should not raise any errors.

## Environment

 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): MacOS
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Python version: Python 3.7.4

cc @ezyang

**Comment by void32** [comment]
I'm experiencing the same problem:
    PyTorch Version (e.g., 1.0): 1.4.0
    OS (e.g., Linux): Linux
    How you installed PyTorch (conda, pip, source): pip
    Python version: Python 3.7.6

**Comment by mruberry** [comment]
I just tried this and got a torrent of errors, which isn't especially surprising. If you're only receiving this error you may not be setting your MYPYPATH environment variable properly. Regardless, you probably want to ignore PyTorch imports when using MyPy. 

Please see the MyPy documentation for more on its (complex) import semantices.

**Comment by ezyang** [comment]
Should be an easy enough fix, just need to add stubs for AdamW. Add me as reviewer on a PR if you fix this.

**Comment by prajjwal1** [comment]
Hi,
I can work on this. 
I am seeing stub file defined for AdamW [here](https://github.com/pytorch/pytorch/blob/master/torch/optim/adamw.pyi). It is also imported in the __init__ [here](https://github.com/pytorch/pytorch/blob/master/torch/optim/__init__.pyi). Could you please provide a pointer on what needs to be done ?

**Comment by prajjwal1** [comment]
@ezyang I checked locally about this change. Mypy didn't throw any error after making the edit. Could you please have a look ?

**Comment by prajjwal1** [comment]
@yf225 You can close this issue as it has been resolved in [#34299](https://github.com/pytorch/pytorch/pull/34299)

**Comment by vvolhejn** [comment]
I just had the same issue with Pyright:

```
error: "AdamW" is not exported from module "torch.optim" (reportPrivateImportUsage)
```

Fixed by importing `from torch.optim.adamw import AdamW` rather than `from torch.optim import AdamW`.
